llm/server.go

```diff
@@ -841,12 +842,70 @@ func (s *llmServer) Completion(ctx context.Context, req CompletionRequest, fn fu
 	return nil
 }
 
+type EmbedRequest struct {
+	Content []string `json:"content"`
+}
+
+type EmbedResponse struct {
+	Embedding [][]float64 `json:"embedding"`
+}
+
+func (s *llmServer) Embed(ctx context.Context, input []string) ([][]float64, error) {
```

Comment:
We should be able to merge this with `Embedding` below to only have one interface to `server.cpp`

Reply:
do we want to change Embedding to also return `float32`? otherwise the merged interface would take in `any` as input and also return `any` which could be a bit messy